internal/tui/components/chat/editor/editor.go

```diff
@@ -112,7 +113,41 @@ func (m *editorCmp) openEditor(value string) tea.Cmd {
 	if _, err := tmpfile.WriteString(value); err != nil {
 		return util.ReportError(err)
 	}
-	c := exec.CommandContext(context.TODO(), editor, tmpfile.Name())
+
+	// Parse the EDITOR parts to separate the executable from its arguments.
+	// This properly handles EDITORs with args like `zed --wait`.
+	parsed, err := syntax.NewParser().Parse(strings.NewReader(editor), "")
+	if err != nil {
+		return util.ReportError(fmt.Errorf("failed to parse editor command: %w", err))
+	}
+
+	var cmdName string
+	var cmdArgs []string
+	if len(parsed.Stmts) > 0 && parsed.Stmts[0].Cmd != nil {
+		if callExpr, ok := parsed.Stmts[0].Cmd.(*syntax.CallExpr); ok && len(callExpr.Args) > 0 {
+			for i, arg := range callExpr.Args {
+				var argStr string
+				for _, part := range arg.Parts {
+					if lit, ok := part.(*syntax.Lit); ok {
+						argStr += lit.Value
+					}
+				}
+				if i == 0 {
+					cmdName = argStr
+				} else {
+					cmdArgs = append(cmdArgs, argStr)
+				}
+			}
+		}
+	}
```

Comment:
I wonder if we could just run the script directly with `mvdan.cc/sh/interp` to avoid having to manipulate the AST this way. Basically:

1. `cmd := fmt.Sprintf("%s %s", os.Getenv("EDITOR"), filePath)`
2. Run `cmd` with the interpreter

If that doesn't work well, I would at very least extract this logic into a separate function to keep this one as clean as possible.